src/commands/index.rs

```diff
@@ -147,16 +148,27 @@ where
         .and_then(|index| {
             let results: Result<Vec<_>, RedisError> = index
                 .search(&query)?
-                .map(|key| {
+                .filter_map(|key| {
                     let key = ctx.open_key_writable(&key);
-                    let value = match key.get_value::<RedisJSON>(&REDIS_JSON_TYPE)? {
-                        Some(doc) => doc.to_string(&path, Format::JSON)?.into(),
-                        None => RedisValue::None,
-                    };
-                    Ok(value)
+                    key.get_value::<RedisJSON>(&REDIS_JSON_TYPE)
+                        .and_then(|value| {
+                            value
+                                .and_then(|doc| {
+                                    doc.get_values(&path)
+                                        .and_then(|values| {
+                                            values
+                                                .first()
+                                                .map(|v| RedisJSON::serialize(v, Format::JSON)) // Option<Result<String, Error>>
```

Comment:
I think that adding comments with type annotations is not a good idea: When the code will change, they will get out of sync. If you think the annotations are important, move them out of the comments as explicit annotations in the code (personally I don't see them adding any value -- this is something the IDE can show automatically if desired).

Reply:
It just helped me to understand what is going on

Reply:
removed